src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -77,17 +118,190 @@ public String getScript() {
+        return PARSER.parse(config, removeMatchedEntries);
+    }
+
+    public static Script parse(XContentParser parser) throws IOException {
+        return PARSER.parse(parser);
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((lang == null) ? 0 : lang.hashCode());
+        result = prime * result + ((params == null) ? 0 : params.hashCode());
+        result = prime * result + ((script == null) ? 0 : script.hashCode());
+        result = prime * result + ((type == null) ? 0 : type.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Script other = (Script) obj;
+        if (lang == null) {
+            if (other.lang != null)
+                return false;
+        } else if (!lang.equals(other.lang))
+            return false;
+        if (params == null) {
+            if (other.params != null)
+                return false;
+        } else if (!params.equals(other.params))
+            return false;
+        if (script == null) {
+            if (other.script != null)
+                return false;
+        } else if (!script.equals(other.script))
+            return false;
+        if (type != other.type)
+            return false;
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        return "[script: " + script + ", type: " + type.getParseField().getPreferredName() + ", lang: " + lang + ", params: " + params
+                + "]";
+    }
+
+    private static class ScriptParser extends AbstractScriptParser<Script> {
+
+        @Override
+        protected Script createSimpleScript(XContentParser parser) throws IOException {
+            if (parser.currentToken() == XContentParser.Token.VALUE_STRING) {
+                return new Script(parser.text());
+            } else {
+                throw new ScriptParseException("expected a string value for field [{}], but found [{}]", parser.currentName(),
+                        parser.currentToken());
+            }
+        }
+
+        @Override
+        protected Script createScript(String script, ScriptType type, String lang, Map<String, Object> params) {
+            return new Script(script, type, lang, params);
+        }
+
+        @Override
+        protected String parseInlineScript(XContentParser parser) throws IOException {
+            return parser.text();
+        }
+    }
+
+    public static interface ScriptField {
+        ParseField SCRIPT = new ParseField("script");
+        ParseField LANG = new ParseField("lang");
+        ParseField PARAMS = new ParseField("params");
+    }
+
+    public static class ScriptParseException extends ElasticsearchException {
```

Comment:
this new class is going to trigger errors too if we try to serialize it to an older node